examples/validation-app/src/interceptors/validate-phone-num.interceptor.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/example-validation-app
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  bind,
+  Interceptor,
+  InvocationContext,
+  InvocationResult,
+  Provider,
+  ValueOrPromise,
+} from '@loopback/context';
+import {CoffeeShop} from '../models';
+
+/**
+ * This class will be bound to the application as an `Interceptor` during
+ * `boot`
+ */
+@bind({tags: {key: ValidatePhoneNumInterceptor.BINDING_KEY}})
+export class ValidatePhoneNumInterceptor implements Provider<Interceptor> {
+  static readonly BINDING_KEY = `interceptors.${ValidatePhoneNumInterceptor.name}`;
+
+  /*
+  constructor() {}
+  */
+
+  /**
+   * This method is used by LoopBack context to produce an interceptor function
+   * for the binding.
+   *
+   * @returns An interceptor function
+   */
+  value() {
+    return this.intercept.bind(this);
+  }
+
+  /**
+   * The logic to intercept an invocation
+   * @param invocationCtx - Invocation context
+   * @param next - A function to invoke next interceptor or the target method
+   */
+  async intercept(
+    invocationCtx: InvocationContext,
+    next: () => ValueOrPromise<InvocationResult>,
+  ) {
+    // Add pre-invocation logic here
+    const coffeeShop: CoffeeShop = new CoffeeShop();
+    if (invocationCtx.methodName === 'create')
```

Comment:
hmm, how about using [method level interceptor](https://loopback.io/doc/en/lb4/Interceptors.html#method-level-interceptors) to generate the instance?
I think the situation is similar to what I did in the access control example's method level voter: before executing the validation/authorization logic, we need do some pre-work based on method.

And I am a bit confused why have to create a new model instance first? Wouldn't the data from request already a model instance?

(Considering this example aims to demo validation. The interceptor details doesn't can be improved in a 2nd round :p)

Reply:
@emonddr had the same question about instantiating the CoffeeShop instance, but I couldn't find a good alternatives. See discussion: https://github.com/strongloop/loopback-next/pull/4874/files#r392316159. Please let me know if you have a better solution! Thanks.

Reply:
For method level interceptor, I'm thinking the interceptor should be applied to multiple methods in the CoffeeShopController (create, update, updateById, patch), so that's why i'm using class level interceptor.

Reply:
You can still apply the decorator at relevant methods. That might be more explicit.

Reply:
@dhmlau The method level interceptor can be separated from the global one, e.g. `create` method has an endpoint level interceptor which adds the coffeeshop instance to the invocationContext, and the controller level interceptor does the validation. 

I am good with the current implementation to land this PR, we can always discuss and improve after it.

Reply:
Talked to @jannyHou. She was thinking to have the class level interceptor to validate the instance and retrieve the instance based on the method name in the method level interceptor. But it's good with my current implementation.